### PR TITLE
[BUG][NRPTI-1233] Amalgamated permits not showing in BCMI

### PR DESCRIPTION
**Pull request by sggerard**
# Description

This PR includes the following proposed change(s):
* Support for new permit code 'ALG' in both CORE cron job and front end
* Migration to update existing BCMI collections with type of 'ALG'
* Additional minor unrelated spelling and linting fixes

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name: Bug
 about: An undesirable behaviour that needs correction
 title: ''
-labels: Spike
+labels: Bug
```
Comment (acatchpole) [inline .github/ISSUE_TEMPLATE/bug.md:5]: unrelated minor fix to a github issue template

File: api/src/controllers/mine-controller.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ exports.protectedPut = async function (args, res, next) {
 
   const sanitizedObj = PutUtils.validateObjectAgainstModel(MineBCMI, incomingObj);
 
-  if (!sanitizedObj || sanitizedObj === {}) {
+  if (!sanitizedObj || Object.keys(sanitizedObj).length === 0) {
```
Comment (acatchpole) [inline api/src/controllers/mine-controller.js:75]: unrelated minor fix. as js compares objects by reference, "sanitizedObj === {}" is always false.

**Comment by acatchpole** [comment]
sonarcloud is failing due to some var declarations in the migration template and a function we did not write having too high of a "cognitive load".

File: angular/projects/admin-nrpti/src/app/mines/mines-record-add/mines-record-add.component.html [hunk 3]
```diff
@@ -46,7 +46,9 @@
           </label>
           <select name="typeCode" id="typeCode" formControlName="typeCode" class="form-control">
             <option *ngFor="let typeCode of permitTypes" [ngValue]="typeCode">
-              {{ typeCode === 'OGP' ? 'Original Permit' : 'Permit Amendment' }}
+              {{
```
Comment (fergmac) [inline angular/projects/admin-nrpti/src/app/mines/mines-record-add/mines-record-add.component.html:49]: I prefer to avoid nested ternary operators. I find they can be confusing to reason about and prefer the clarity of something along the following lines. Not a necessary change though. You'd have to pull it out into a function.

`
const acceptedTypeCodes = ['OGP', 'ALG']

if (acceptedTypeCodes.includes(typeCode)) return 'Permit Amendment';

if (typeCode === 'OGP') return 'Original Permit';

if (typeCode === 'ALG') return 'Amalgamated Permit';
`

File: angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.html [hunk 4]
```diff
@@ -68,7 +68,9 @@ <h2 class="border-0 mb-0">Edit record</h2>
           <label for="typeCode">Permit Type</label>
           <select name="typeCode" id="typeCode" formControlName="typeCode" class="form-control">
             <option *ngFor="let typeCode of permitTypes" [ngValue]="typeCode">
-              {{ typeCode === 'OGP' ? 'Original Permit' : 'Permit Amendment' }}
+              {{
```
Comment (fergmac) [inline angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.html:71]: We are repeating here so it could make sense to pull this out into a function referenced above.

File: angular/projects/admin-nrpti/src/app/mines/mines-records-detail/mines-records-detail.component.html [hunk 5]
```diff
@@ -42,6 +42,8 @@ <h2 class="border-0 mb-0">Record Information</h2>
                 ? 'Original Permit'
                 : record && record.typeCode && record.typeCode === 'AMD'
                 ? 'Permit Amendment'
+                : record && record.typeCode && record.typeCode === 'ALG'
+                ? 'Amalgamated Permit'
```
Comment (fergmac) [inline angular/projects/admin-nrpti/src/app/mines/mines-records-detail/mines-records-detail.component.html:46]: This ternary could also be addressed if you do decide to pull out a function and go the route of early returns.

File: api/src/integrations/core/permit-utils.js [hunk 6]
```diff
@@ -109,6 +109,17 @@ class Permits extends BaseRecordUtils {
 
     return permits;
   }
+
+  getPermitType(typeCode) {
+    switch(typeCode){
+      case 'OGP':
+        return 'Permit';
```
Comment (fergmac) [inline api/src/integrations/core/permit-utils.js:116]: We reference 'Permit' here and 'Original Permit' for the same case it would be best to stick to one name but there may be a reason for that.

Also is there an opportunity to refactor this switch into the function suggested above?

**Review by fergmac — DISMISSED** [review]

**Review by acatchpole — APPROVED** [review]
